platform/lib/build/samplesBuilder.js

```diff
@@ -80,128 +80,90 @@ class SamplesBuilder {
-   * @param {boolean} watch  Watch for changes
-   * @return {Promise<void>} Resolves when build is done
+   * Executes all async operations that need to be done
+   * before the samples builder is operational
+   * @return {Promise<void>} Resolves when bootstrapping is done
    */
-  async build(watch) {
-    this._formatTransform = await formatTransform.getInstance();
-
-    // Configure cache
-    this._cache[STORY_EMBED_SNIPPET] = await readFileAsync(STORY_EMBED_SNIPPET);
-    this._cache[ADS_EMBED_TEMPLATE] = (await readFileAsync(ADS_EMBED_TEMPLATE)).toString();
-    this._cache.categories = {};
+  async _bootstrap() {
+    if (this._bootstrapped) {
+      return;
+    }
 
-    // If samples should be rebuild (due to architectural changes for example)
-    // then you should be able to clean the sample build destinations
-    if (config.options['clean-samples'] === true) {
-      this._log.info('Cleaning sample destinations for rebuild ...');
-      del.sync([
-        // Clean old structure with multiple collections
-        utils.project.absolute(`/pages/${DOCUMENTATION_POD_PATH}/../**/*`),
-        '!' + utils.project.absolute(`/pages/${DOCUMENTATION_POD_PATH}/../`),
-        '!' + utils.project.absolute(`/pages/${DOCUMENTATION_POD_PATH}/../_blueprint.yaml`),
-        '!' + utils.project.absolute(`/pages/${DOCUMENTATION_POD_PATH}/../index.html`),
-
-        `${DOCUMENTATION_DEST}/**/*`,
-        `!${DOCUMENTATION_DEST}`,
-        `!${DOCUMENTATION_DEST}/_blueprint.yaml`,
-
-        `${PREVIEW_DEST}/**/*`,
-        `!${PREVIEW_DEST}`,
-        `!${PREVIEW_DEST}/_blueprint.yaml`,
-
-        `${SOURCE_DEST}`,
-        `${EMBED_DEST}`,
-        CACHE_DEST,
+    this._log.info('Bootstrapping ...');
+    await Promise.all([
+      readFileAsync(STORY_EMBED_SNIPPET).then((template) => {
+        this._cache[STORY_EMBED_SNIPPET] = template.toString();
+        this._log.success('Loaded story embed template.');
+      }),
+
+      readFileAsync(ADS_EMBED_TEMPLATE).then((template) => {
+        this._cache[ADS_EMBED_TEMPLATE] = template.toString();
+        this._log.success('Loaded ads embed template.');
+      }),
+
+      formatTransform.getInstance().then((instance) => {
+        this._formatTransform = instance;
+        this._log.success('Created format transformer instance.');
+      }).catch((e) => {
+        this._log.warn('Could not create format transformer instance. Samples will only be build for their original format.');
+      }),
+
+      del([
+        `${DOCUMENTATION_DEST}/**/*.html`,
+        `${DOCUMENTATION_DEST}/**/*.json`,
+        `${PREVIEW_DEST}/**/*.html`,
+
+        SOURCE_DEST,
+        EMBED_DEST,
+
+        SITEMAP_DEST,
+        COMPONENT_SAMPLES_DEST,
       ], {
         'force': true,
-      });
-    }
+      }).then(() => {
+        this._log.success('Cleaned destination paths.');
+      })
+    ]);
 
-    if (!watch && config.isDevMode() && module.parent) {
+    this._bootstrapped = true;
```

Comment:
At a glance it might look like this but all of the async operations are awaited with `await Promise.all` above and `this._bootstrapped` will only be true after they've completed.

Suggestions on how to improve readability here?